### PR TITLE
ref(types): Update SelectValue type

### PR DESCRIPTION
**Pull request by vuluongj20**
Include more optional props (e.g. `leadingItems`, `tooltipOptions` - used in `form/selectOption.tsx`) in the `SelectValue` type, so we don't have to manually extend `SelectValue` when using these props.

### CODE REVIEW

File: static/app/types/core.tsx [hunk 1]
```diff
@@ -5,6 +5,7 @@
  * or used in multiple views.
  */
 import type {getInterval} from 'sentry/components/charts/utils';
+import type {InternalTooltipProps} from 'sentry/components/tooltip';
```
Comment (vuluongj20) [inline static/app/types/core.tsx:8]: not sure why this is called `internal` when it's still exported - are we not supposed to directly import it?

Comment (evanpurkhiser) [reply static/app/types/core.tsx:8]: @JonasBa

File: static/app/types/core.tsx [hunk 2]
```diff
@@ -47,8 +48,14 @@ export type Writable<T> = {-readonly [K in keyof T]: T[K]};
 export type SelectValue<T> = {
   label: string | number | React.ReactElement;
   value: T;
+  details?: React.ReactNode;
```
Comment (evanpurkhiser) [inline static/app/types/core.tsx:51]: Hmmmmm, shouldn't all these things be part of the label? This change seems a bit suspect.

Comment (vuluongj20) [reply static/app/types/core.tsx:51]: I added `details` as a prop so its content can be rendered inside [a styled `<Details>` element](https://github.com/getsentry/sentry/blob/vl/select-option-type/static/app/components/forms/selectOption.tsx#L55), so they would automatically have smaller and lighter text.

Screen Shot 2022-05-05 at 2 17 55 PM

It's just useful for maintaining visual consistency between selectors. If you want to add supplementary but non-essential text, just put it in the `details` prop and don't worry about styling it.

**Review by scttcper — APPROVED** [review]
seems okay to me

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**